File: 6405341/Week 1/FactoryMethodPattern/Test_Factory_Method.java
package FactoryMethodPattern;

public class Test_Factory_Method {
    public static void main(String[] args) {
        Document_Factory pdfFactory = new Pdf_Factory();
        Document pdfDoc = pdfFactory.createDocument();
        pdfDoc.open();

        Document_Factory wordFactory = new Word_Factory();
        Document wordDoc = wordFactory.createDocument();
        wordDoc.open();

        Document_Factory excelFactory = new Excel_Factory();
        Document excelDoc = excelFactory.createDocument();
        excelDoc.open();

        System.out.println("\nDone by Superset ID: 6405341\n");
    }
}

File: 6405341/Week 1/FactoryMethodPattern/Test_Factory_Method.cs
using System;

namespace FactoryMethodPattern
{
    class Test_Factory_Method
    {
        static void Main(string[] args)
        {
            DocumentFactory pdfFactory = new PdfFactory();
            Document pdfDoc = pdfFactory.CreateDocument();
            pdfDoc.Open();

            DocumentFactory wordFactory = new WordFactory();
            Document wordDoc = wordFactory.CreateDocument();
            wordDoc.Open();

            DocumentFactory excelFactory = new ExcelFactory();
            Document excelDoc = excelFactory.CreateDocument();
            excelDoc.Open();

            Console.WriteLine("\nDone by Superset ID: 6405341\n");
        }
    }
}
